tests/bluetooth/tester/src/audio/btp_bap_broadcast.c

```diff
@@ -361,14 +427,115 @@ uint8_t btp_bap_broadcast_source_setup(const void *cmd, uint16_t cmd_len,
+		return BTP_STATUS_FAILED;
+	}
+
+	err = bt_bap_broadcast_source_get_base(source->bap_broadcast, &base_buf);
+	if (err != 0) {
+		LOG_DBG("Failed to get encoded BASE: %d\n", err);
+		return BTP_STATUS_FAILED;
+	}
 
+	per_ad = &per_adv_local[source->source_id][0];
+	per_ad->type = BT_DATA_SVC_DATA16;
+	per_ad->data_len = base_buf.len;
+	per_ad->data = base_buf.data;
+	err = tester_gap_padv_set_data(source->ext_adv, per_ad, 1);
+	if (err != 0) {
+		return BTP_STATUS_FAILED;
+	}
+
+	rp->gap_settings = gap_settings;
+	sys_put_le24(broadcast_id, rp->broadcast_id);
+	*rsp_len = sizeof(*rp) + 1;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+uint8_t btp_bap_broadcast_source_setup_v2(const void *cmd, uint16_t cmd_len,
+					  void *rsp, uint16_t *rsp_len)
+{
+	struct bt_le_per_adv_param per_adv_param =
+		*BT_LE_PER_ADV_PARAM(BT_GAP_MS_TO_PER_ADV_INTERVAL(150),
+				     BT_GAP_MS_TO_PER_ADV_INTERVAL(150), BT_LE_PER_ADV_OPT_NONE);
+	/* Zephyr Controller works best while Extended Advertising interval is a multiple
+	 * of the ISO Interval minus 10 ms (max. advertising random delay). This is
+	 * required to place the AUX_ADV_IND PDUs in a non-overlapping interval with the
+	 * Broadcast ISO radio events.
+	 */
+	struct bt_le_adv_param ext_adv_param =
+		*BT_LE_ADV_PARAM(BT_LE_ADV_OPT_EXT_ADV, BT_GAP_MS_TO_ADV_INTERVAL(140),
+				 BT_GAP_MS_TO_ADV_INTERVAL(140), NULL);
+	int err;
+	struct bt_audio_codec_cfg codec_cfg;
+	const struct btp_bap_broadcast_source_setup_v2_cmd *cp = cmd;
+	struct btp_bap_broadcast_source_setup_v2_rp *rp = rsp;
+	uint32_t broadcast_id = sys_get_le24(cp->broadcast_id);
+
+	struct btp_bap_broadcast_local_source *source;
+
+	uint32_t gap_settings = BIT(BTP_GAP_SETTINGS_DISCOVERABLE) |
+				BIT(BTP_GAP_SETTINGS_EXTENDED_ADVERTISING);
+
+	NET_BUF_SIMPLE_DEFINE(ad_buf, BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE);
+	NET_BUF_SIMPLE_DEFINE(base_buf, 128);
+
+	/* Broadcast Audio Streaming Endpoint advertising data */
+	struct bt_data base_ad[2];
+	struct bt_data *per_ad;
+
+	LOG_DBG("");
+
+	memset(&codec_cfg, 0, sizeof(codec_cfg));
+	codec_cfg.id = cp->coding_format;
+	codec_cfg.vid = cp->vid;
+	codec_cfg.cid = cp->cid;
+	codec_cfg.data_len = cp->cc_ltvs_len;
+	memcpy(codec_cfg.data, cp->cc_ltvs, cp->cc_ltvs_len);
+
+	source = btp_bap_broadcast_local_source_allocate(broadcast_id);
+	if (source == NULL) {
+		LOG_DBG("No more free local source items");
+		return BTP_STATUS_FAILED;
+	}
+
+	source->qos.phy = BT_BAP_QOS_CFG_2M;
+	source->qos.framing = cp->framing;
+	source->qos.rtn = cp->retransmission_num;
+	source->qos.latency = sys_le16_to_cpu(cp->max_transport_latency);
+	source->qos.interval = sys_get_le24(cp->sdu_interval);
+	source->qos.pd = sys_get_le24(cp->presentation_delay);
+	source->qos.sdu = sys_le16_to_cpu(cp->max_sdu);
+
+	source->stream_count = cp->subgroups * cp->streams_per_subgroup;
+
+	err = setup_broadcast_source(cp->streams_per_subgroup, cp->subgroups, source, &codec_cfg);
```

Comment:
this free source on error?

Reply:
No, but the test will fail thus resetting the entire system. So do we need to free source in this case?

Reply:
This is the same as in v1, and there are also multiple places in the code where things are not freed just before sending `BTP_STATUS_FAILED`